app/org/sagebionetworks/bridge/services/CompoundActivityDefinitionService.java

```diff
@@ -111,7 +111,8 @@ public CompoundActivityDefinition updateCompoundActivityDefinition(StudyIdentifi
     
     private void checkConstraintViolations(StudyIdentifier studyId, String taskId) {
         // Cannot delete a definition if it is referenced in any schedule plan.
-        List<SchedulePlan> plans = schedulePlanService.getSchedulePlans(ClientInfo.UNKNOWN_CLIENT, studyId);
+        // If you delete a schedule plan, than you can delete a compound activity definition pointing to it.
+        List<SchedulePlan> plans = schedulePlanService.getSchedulePlans(ClientInfo.UNKNOWN_CLIENT, studyId, false);
```

Comment:
As per offline conversation, to be consistent, if we're logically deleting, it's fine if we don't check logically deleted schedule plans. But if we're physically deleting, we should make sure logically deleted schedule plans don't reference this, since they can still be undeleted. (Here and in SurveyService.)